features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt

```diff
@@ -5036,6 +5036,85 @@ internal class RumViewScopeTest {
         assertThat(result).isSameAs(testedScope)
     }
 
+    @Test
+    fun `𝕄 send event 𝕎 handleEvent(AddError) on active view { error fingerprint attribute }`(
+        @Forgery source: RumErrorSource,
+        @Forgery throwable: Throwable,
+        @StringForgery stacktrace: String,
+        @StringForgery fingerprint: String,
+        forge: Forge
+    ) {
+        // Given
+        testedScope.activeActionScope = mockActionScope
+        val mockAttributes = forge.exhaustiveAttributes(excludedKeys = fakeAttributes.keys)
+        val fullAttributes = mockAttributes.toMutableMap().apply {
+            put(RumAttributes.ERROR_FINGERPRINT, fingerprint)
+        }
+        val throwableMessage = throwable.message
+        check(!throwableMessage.isNullOrBlank()) {
```

Comment:
better to use the `assumeTrue` here, I don't think we want to fail the test if test setup is not good.

but anyway, is this needed here? what changes if message is null (it is never possible by looking on `ThrowableForgeryFactory`) or blank?

Reply:
Since `throwable.message` is `Stirng?`, `check` performs the null check that allows us to no not perform the force dereference further down.  I doesn't seem like `assumeTrue` does that.

Reply:
yes, `check` is smart-cast friendly. however, we don't want test to fail if pre-condition is not satisfied, and `assumeTrue` will allow test to continue.

Speaking about this scenario, I think it is just easier to generate non-null message manually using `@StringForgery fakeMessage: String`, and then things will be simpler.